File: soms_mobile_app/windows/lib/models/fine.dart
class Fine {
  const Fine({
    required this.id,
    required this.userId,
    this.studentName,
    this.studentIdNumber,
    required this.violationType,
    required this.reason,
    required this.amount,
    required this.status, // unpaid | paid | waived
    required this.issuedAt,
    this.eventTitle,
    this.sessionType,
  });

  final int id;
  final int userId;
  final String? studentName;
  final String? studentIdNumber;
  final String violationType;
  final String reason;
  final double amount;
  final String status;
  final String issuedAt;
  final String? eventTitle;
  final String? sessionType;

  bool get isUnpaid => status == 'unpaid';
  bool get isPaid => status == 'paid';
  bool get isWaived => status == 'waived';

  factory Fine.fromJson(Map<String, dynamic> json) {
    final user = json['user'] as Map<String, dynamic>?;
    final event = json['event'] as Map<String, dynamic>?;
    final session = json['event_session'] as Map<String, dynamic>?;
    return Fine(
      id: json['id'] as int,
      userId: json['user_id'] as int,
      studentName: user?['name'] as String?,
      studentIdNumber: user?['student_id'] as String?,
      violationType: json['violation_type'] as String? ?? '',
      reason: json['reason'] as String? ?? '',
      amount: double.tryParse('${json['amount']}') ?? 0,
      status: json['status'] as String? ?? 'unpaid',
      issuedAt: json['issued_at'] as String? ?? '',
      eventTitle: event?['title'] as String?,
      sessionType: session?['session_type'] as String?,
    );
  }
}

File: soms/app/Models/Fine.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Fine extends Model
{
    protected $fillable = [
        'user_id',
        'event_id',
        'event_day_id',
        'event_session_id',
        'violation_type',
        'reason',
        'amount',
        'status',
        'issued_at',
        'cleared_by',
        'cleared_at',
    ];

    protected $casts = [
        'amount'     => 'decimal:2',
        'issued_at'  => 'datetime',
        'cleared_at' => 'datetime',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function event(): BelongsTo
    {
        return $this->belongsTo(Event::class);
    }

    public function eventDay(): BelongsTo
    {
        return $this->belongsTo(EventDay::class);
    }

    public function eventSession(): BelongsTo
    {
        return $this->belongsTo(EventSession::class);
    }

    public function clearedBy(): BelongsTo
    {
        return $this->belongsTo(User::class, 'cleared_by');
    }

    public function scopeUnpaid($query)
    {
        return $query->where('status', 'unpaid');
    }

    public function scopeWaived($query)
    {
        return $query->where('status', 'waived');
    }

    public function scopeByEvent($query, $eventId)
    {
        return $query->where('event_id', $eventId);
    }
}
